components/motor/gpio/motor_encoder.go

```diff
@@ -256,13 +257,41 @@ func (m *EncodedMotor) goForMath(ctx context.Context, rpm, revolutions float64)
 // If revolutions is 0, this will run the motor at rpm indefinitely
 // If revolutions != 0, this will block until the number of revolutions has been completed or another operation comes in.
 func (m *EncodedMotor) GoFor(ctx context.Context, rpm, revolutions float64, extra map[string]interface{}) error {
-	m.opMgr.CancelRunning(ctx)
 	ctx, done := m.opMgr.New(ctx)
 	defer done()
 
 	goalPos, goalRPM, direction := m.goForMath(ctx, rpm, revolutions)
 
-	return m.goForInternal(ctx, goalRPM, goalPos, direction)
+	if err := m.goForInternal(ctx, goalRPM, goalPos, direction); err != nil {
+		return err
+	}
+
+	if revolutions == 0 {
+		return nil
+	}
+
+	positionReached := func(ctx context.Context) (bool, error) {
+		var errs error
+		pos, posErr := m.position(ctx, extra)
+		errs = multierr.Combine(errs, posErr)
+		if (direction == 1 && pos >= goalPos) || (direction == -1 && pos <= goalPos) {
+			stopErr := m.Stop(ctx, extra)
+			errs = multierr.Combine(errs, stopErr)
+			return true, errs
+		}
+		return false, errs
+	}
+	err := m.opMgr.WaitForSuccess(
+		ctx,
+		10*time.Millisecond,
+		positionReached,
+	)
+	// Ignore the context canceled error - this occurs when the motor is stopped
+	// at the beginning of goForInternal
+	if !errors.Is(err, context.Canceled) {
```

Comment:
The comment confuses me: is this saying that the `WaitForSuccess` call always returns early with errors that should be ignored? Does that mean we never actually wait the right amount? I don't see anywhere in `goForInternal` that would cancel the context (it doesn't even use it!), so I don't even see where the error comes from. Are you sure about this?

Reply:
just confirmed, yes this is still necessary.

Reply:
oh I didn't read the comment yeah that's wrong. it's when we call stop when we reach the goal position that we get a context canceled sorry about the confusion

Reply:
well wait no it can happen in goForInternal too... I think it's line 308, `m.rpmMonitorDone()` which also happens in `Stop` which we call when we reach the goal position

Reply:
just updated the comment